script.go

```diff
@@ -762,6 +764,30 @@ func (p *Pipe) SetError(err error) {
 	p.err = err
 }
 
+// setStderr sets the stderr writer on the pipe. This field
+// is protected by a mutex since stderr is accessed inside a
+// goroutine from [Pipe.Exec].
+func (p *Pipe) setStderr(stderr io.Writer) {
+	if p.mu == nil { // uninitialised pipe
+		return
+	}
+	p.mu.Lock()
+	defer p.mu.Unlock()
+	p.stderr = stderr
+}
+
+// getStderr obtains the stderr writer on the pipe. This field
+// is protected by a mutex since stderr is accessed inside a
+// goroutine from [Pipe.Exec].
+func (p *Pipe) getStderr() io.Writer {
+	if p.mu == nil { // uninitialised pipe
+		return nil
+	}
+	p.mu.Lock()
```

Comment:
```suggestion
	p.mu.RLock()
```

I think we only need a read lock here.

Reply:
@bitfield we can't do this since we're using a `sync.Mutex` instead of a `sync.RWMutex`

Reply:
This needs a little more thought, doesn't it? If we're just _retrieving_ the `stderr` writer from the `Pipe` struct, then to be sure we only need a read lock (and you've answered your own question, in a way: the solution would be to change the existing `sync.Mutex to a `sync.RWMutex`).

But the question arises whether it's safe for multiple concurrent `Exec`s to all write to the same writer. We can't know that it is. On the other hand, if we were to enforce serialization with a separate mutex for this error writer, then we'd lose the concurrency.

On balance I think we should let the user worry about that, don't you? After all, the configured stderr is overwhelmingly likely to be either `os.Stderr` or some file, in which case you might end up with interleaved writes looking messy, but no actual crashes.

As far as the `p.mu` goes, I think we can keep it as a write-lock for now: it doesn't seem like this is in the hot path of anything.

Reply:
Sounds good! Thank you for the detailed explanation @bitfield